script/check_changelog.py

```diff
@@ -0,0 +1,105 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Small script to check the changelog. Used by 'changelog.yml' and pre-commit.
+
+If no issue number is provided we only check that proper formatting is respected."""
+
+from __future__ import annotations
+
+import argparse
+import re
+import sys
+from collections.abc import Iterator
+from pathlib import Path
+from re import Pattern
+
+ISSUE_NUMBER_PATTERN = re.compile(r"#\d{1,5}")
```

Comment:
Can't we reuse this pattern in the other patterns?

```suggestion
ISSUE_NUMBER_PATTERN = re.compile(r"#\d+")
```

Reply:
I see you're an optimist 😄

Reply:
Still, let's reuse the pattern so if we ever need to change it that's just in one place.

Reply:
Done in [9a60d73](https://github.com/PyCQA/pylint/pull/6735/commits/9a60d73c49cc293a666f5ed32b3dc685e19f8872)  with some bonus modularity